File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/repository/MemberRepository.java
package com.kakao.mobility.kmcoupon.repository;

import com.kakao.mobility.kmcoupon.domain.member.Member;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long> {
    Optional<Member> findByEmail(String email);
}

File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/repository/MemberRepository.kt
package com.kakao.mobility.kmcoupon.repository

import com.kakao.mobility.kmcoupon.domain.member.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByEmail(email: String): Optional<Member>
}